src/backend/metal/src/device.rs

```diff
@@ -1209,7 +1217,7 @@ impl hal::Device<Backend> for Device {
         pipeline.set_input_primitive_topology(primitive_class);
 
         // Vertex shader
-        let (vs_lib, vs_function, _) = self.load_shader(
+        let (vs_lib, vs_function, _, enable_rasterization) = self.load_shader(
```

Comment:
It's a bit strange – SPIRV-Cross (1) accepts both a flag to specify whether rasterization should be enabled, and also (2) informs us whether or not rasterization was actually enabled on the generated output.

For (1), I haven't changed the input to flag to indicate whether rasterization *should* be enabled or not yet (though we should probably do this soon), because I was just working through the specialization constant cases for now.

The reasoning for (2) is because MSL has some restrictions on when the return type must be `void`, see https://github.com/KhronosGroup/SPIRV-Cross/pull/644 In the CTS, there are a bunch of cases which write directly to buffers within the vertex shader, so SPIRV-Cross forces the return type to `void`, and we only can find this out by querying SPIRV-Cross and deciding whether or not to disable rasterization for the pipeline.